### PR TITLE
Rename patchwork-client-colors to patchwork-events-rendering and implement TextureStitchEvent

### PR DESCRIPTION
**Pull request by famous1622**
As discussed on the Discord, the rename is to have one module for things like colors, model baking, etc. I've implemented TextureStitchEvent, but someone who understands rendering better than me should probably implement model loading and the bake event.

### CODE REVIEW

**Comment by rikka0w0** [comment]
I'm going to impl the IModelLoader and IModelGeometry once I finish my custom registery impl

File: patchwork-events-rendering/src/main/java/net/minecraftforge/client/event/TextureStitchEvent.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+		this.map = map;
+	}
+
+	public SpriteAtlasTexture getMap() {
+		return map;
+	}
+
+	/**
+	 * Fired when the TextureMap is told to refresh it's stitched texture.
+	 * Called before the {@link net.minecraft.client.texture.SpriteAtlasTexture} are loaded.
```
Comment (coderbot16) [inline patchwork-events-rendering/src/main/java/net/minecraftforge/client/event/TextureStitchEvent.java:42]: No need to fully qualify, it's already imported
```suggestion
	 * Called before the {@link SpriteAtlasTexture}s are loaded.
```

File: patchwork-events-rendering/src/main/java/net/minecraftforge/client/event/TextureStitchEvent.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	public static class Pre extends TextureStitchEvent {
+		private final Set<Identifier> sprites;
+
+		public Pre(SpriteAtlasTexture map, Set<Identifier> sprites) {
+			super(map);
+			this.sprites = sprites;
+		}
+
+		/**
+		 * Add a sprite to be stitched into the Texture Atlas.
```
Comment (coderbot16) [inline patchwork-events-rendering/src/main/java/net/minecraftforge/client/event/TextureStitchEvent.java:53]:
```suggestion
		 * Add a sprite to be stitched into the texture atlas.
```

File: patchwork-events-rendering/src/main/java/net/minecraftforge/client/event/TextureStitchEvent.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+		 */
+		public boolean addSprite(Identifier sprite) {
+			return this.sprites.add(sprite);
+		}
+	}
+
+	/**
+	 * This event is fired once the texture map has loaded all textures and
+	 * stitched them together. All Icons should have there locations defined
```
Comment (coderbot16) [inline patchwork-events-rendering/src/main/java/net/minecraftforge/client/event/TextureStitchEvent.java:62]:
```suggestion
	 * stitched them together. All sprites should have there locations defined
```

File: patchwork-events-rendering/src/main/java/net/patchworkmc/mixin/event/render/MixinSpriteAtlasTexture.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import net.minecraft.client.texture.SpriteAtlasTexture;
+import net.minecraft.resource.ResourceManager;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.profiler.Profiler;
+
+import net.patchworkmc.impl.event.render.RenderEvents;
+
+@Mixin(SpriteAtlasTexture.class)
+public class MixinSpriteAtlasTexture {
+	@Inject(method = "stitch", at = @At(value = "INVOKE", shift = At.Shift.AFTER, target = "Lnet/minecraft/util/profiler/Profiler;swap(Ljava/lang/String;)V", ordinal = 0), locals = LocalCapture.CAPTURE_FAILHARD)
```
Comment (coderbot16) [inline patchwork-events-rendering/src/main/java/net/patchworkmc/mixin/event/render/MixinSpriteAtlasTexture.java:40]: Alternatively, you could inject before `loadSprites`, that's probably more stable than a profiler call. I think either way is acceptable, though.

File: patchwork-events-rendering/src/main/java/net/minecraftforge/client/event/TextureStitchEvent.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+	public TextureStitchEvent(SpriteAtlasTexture map) {
+		this.map = map;
+	}
+
+	public SpriteAtlasTexture getMap() {
+		return map;
+	}
+
+	/**
+	 * Fired when the {@link SpriteAtlasTexture} is told to refresh it's stitched texture.
```
Comment (coderbot16) [inline patchwork-events-rendering/src/main/java/net/minecraftforge/client/event/TextureStitchEvent.java:41]: :bike: :house:
```suggestion
	 * Fired when the {@link SpriteAtlasTexture} is told to refresh its stitched texture.
```

**Pull request merged**